src/Composer/Repository/PathRepository.php

```
@@ -97,36 +101,39 @@ protected function initialize()
     {
         parent::initialize();
 
-        $path = $this->getPath();
-        $composerFilePath = $path.'composer.json';
-        if (!file_exists($composerFilePath)) {
-            throw new \RuntimeException(sprintf('No `composer.json` file found in path repository "%s"', $path));
```

you should keep throwing an exception when none of the path contains a composer.json IMO. It makes it much easier to debug issues in the repository configuration rather than silently ignoring the path